drivers/hmis/app/models/hmis/strict_decimal.rb

```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+class Hmis::StrictDecimal < ActiveModel::Type::Decimal
+  def cast(value)
+    return if value.nil?
+
+    raise ArgumentError, "Invalid decimal value: #{value.inspect}" if value.is_a?(String) && value !~ /\A[+-]?\d+(\.\d+)?\z/
```

Comment:
There are a couple cases where this became too strict I think:

old behavior: `"500."`=>`500.0`
new behavior: `"500."`=> raises error

old behavior: `".5"`=>`0.5`
new behavior: `"0.5"`=> raises error

Reply:
Thanks good point. While testing I also noticed it became more strict about booleans. I updated it to allow casting `false` to 0 and `true` to 1 for int cols.